python/pdo/contract/contract.py

```diff
@@ -97,23 +99,35 @@ def __init__(self, code, state, contract_id, creator_id, **kwargs) :
         self.contract_id = contract_id
         self.creator_id = creator_id
         self.extra_data = kwargs.get('extra_data', {})
-
         self.enclave_map = kwargs.get('enclave_map',{})
+        self.set_replication_parameters()
 
     # -------------------------------------------------------
     def set_state_encryption_key(self, enclave_id, encrypted_state_encryption_key) :
         self.enclave_map[enclave_id] = encrypted_state_encryption_key
 
     # -------------------------------------------------------
     def get_state_encryption_key(self, enclave_id) :
-        return self.enclave_map[enclave_id];
+        return self.enclave_map[enclave_id]
 
     # -------------------------------------------------------
     @property
     def short_id(self) :
         assert self.contract_id
         return hex(abs(hash(self.contract_id)))[2:]
 
+
+    # -------------------------------------------------------
+    def set_replication_parameters(self, num_provable_replicas=1, availability_duration=120):
+
+        replication_params = dict()
+        replication_params['max_num_replicas'] = len(self.enclave_map.keys())
+        replication_params['num_provable_replicas'] = num_provable_replicas
+        replication_params['availability_duration'] = availability_duration #seconds
+        replication_params['service_ids'] = self.enclave_map.keys() #we replicate to storage services associated with all provisioned encalves
+
+        self.replication_params = replication_params
```

Comment:
why not just write the previous values directly into self.replication_params?

Reply:
one field is new, but I guess you are right, i can first copy and just add the new field , will do